spec/services/resolve_account_service_spec.rb

```diff
@@ -220,6 +220,8 @@
           return_values << described_class.new.call('foo@ap.example.com')
         rescue ActiveRecord::RecordNotUnique
           fail_occurred = true
+        ensure
+          RedisConfiguration.pool.checkin if Thread.current[:redis]
```

Comment:
shouldn't it be set to `nil` too? also i'm worried the various concurrency tests we have may become moot because of waiting instead for a connection instead of testing race conditions

Reply:
The thing is, `Thread.current` is only accessible from the current thread, so I'm not even sure there's any point setting it to nil at the end of the thread's lifecycle.

Reply:
yeah you're right on this, I was more thinking about the other places where you don't control the thread's lifecycle explicitly, but in those other places you are setting it to `nil` so this is fine.

Still, a bit worried about this possibly rendering threading tests moot if threads end up waiting after each other. But maybe there aren't such tests, I need to take a look.

Reply:
There isn't another place in specs that uses `Thread.new` and accesses Redis

Reply:
Another thing I'm worrying about is `tootctl`, aren't there multithreaded bits there that could depend on redis?